rollout/sync.go

```diff
@@ -541,27 +572,18 @@ func (c *RolloutController) calculateRolloutConditions(roCtx rolloutContext, new
 }
 
 // persistRolloutStatus persists updates to rollout status. If no changes were made, it is a no-op
-func (c *RolloutController) persistRolloutStatus(orig *v1alpha1.Rollout, newStatus *v1alpha1.RolloutStatus, newPause *bool) error {
+func (c *RolloutController) persistRolloutStatus(roCtx rolloutContext, newStatus *v1alpha1.RolloutStatus) error {
+	orig := roCtx.Rollout()
 	specCopy := orig.Spec.DeepCopy()
-	paused := specCopy.Paused
-	if newPause != nil {
-		paused = *newPause
-		specCopy.Paused = *newPause
-	}
-	newStatus.ObservedGeneration = conditions.ComputeGenerationHash(*specCopy)
 
+	roCtx.PauseContext().CalculatePauseStatus(newStatus)
+	newStatus.ObservedGeneration = conditions.ComputeGenerationHash(*specCopy)
```

Comment:
I feel like you don't need specCopy anymore since you can just pass in roCtx.Rollout().Spec

Reply:
👍